src/acp/internal/getDefaultAccessControlThing.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2021 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { createThing, getThing, ThingPersisted } from "../..";
+import type { WithAccessibleAcr } from "../acp";
+import { getAccessControlResource } from "./getAccessControlResource";
+import {
+  DefaultAccessControlName,
+  getDefaultAccessControlUrl,
+} from "./getDefaultAccessControlUrl";
+
+/** @hidden */
+export function getDefaultAccessControlThing(
+  resource: WithAccessibleAcr,
+  name: DefaultAccessControlName
+): ThingPersisted {
+  const acr = getAccessControlResource(resource);
+  const defaultAccessControlUrl = getDefaultAccessControlUrl(resource, name);
+
+  const accessControlThing = getThing(acr, defaultAccessControlUrl);
```

Comment:
Yes, we don't want to use an arbitrary access control, we want to use one of the default Access Controls that by convention the SDK works on: for ACRs, for the Resource, for their members.

Here for example, if we were to edit a random access control, we would use the one that was inherited and happens not to be overridable from the resource, which is a fair take on ACP implementation from a security perspective.

Reply:
Ok, that makes sense. That works from an SDK perspective, but from a spec perspective do you think a client should be able to differenciate inherited access controls from the ones that are editable inline, or is it by design that only the server can make this distinction ?

Reply:
And also, I'm not a big fan on matching on the URL hash fragment. Do you think we could use a GUUID instead of the default names, and add something like `<access-control#guuid> rdf:type <sdk:defaultAccessControl>` ? This way we'd be making exact IRI matching instead on making string manipulations on IRIs.

Reply:
> Ok, that makes sense. That works from an SDK perspective, but from a spec perspective do you think a client should be able to differenciate inherited access controls from the ones that are editable inline, or is it by design that only the server can make this distinction ?

A client could easily discover that the access control was inherited by checking its parent resources. That being said, I have been thinking for a while about introducing an `acp:inheritedAccessControl` to make it easy to identify them in the current ACR. This is a discussion for later.

> And also, I'm not a big fan on matching on the URL hash fragment. Do you think we could use a GUUID instead of the default names, and add something like <access-control#guuid> rdf:type <sdk:defaultAccessControl> ? This way we'd be making exact IRI matching instead on making string manipulations on IRIs.

That would unnecessarily complicate the code and inventing a type is not the best either. I think that ultimately getting rid of default Access Controls all together is what we should aim for. But for the time being, it's ok as is.

Reply:
I don't think it is okay as it is for the reasons I mentioned. I'm not sure I see how changing `DEFAULT_ACCESS_CONTROL` and the other constants into an IRI rather than in a fragment adds that much complexity, can you elaborate ? Since as you said the names are constants, it is just about changing these constants, and changing the way the thing is eventually created. If no name is provided, an arbitrary identifier is already given by the library, and adding a type can be done through the already existing thing builder.

 Also, if we get rid of the default Access controls eventually, removing the triple that gives it a type is easy, while the name cannot be changed.

Reply:
Why would you invent a type and use a GUID to indicate a thing that is meant to be unique and a constant. I want only one constant default Access Control for each of the purposes the SDK facilitates editing for each ACR and I don't want to polute the existing graph.

If I do a GUID with a type, there is no guarantee that the required default access control would be unique. There is also no guarantee that the authorization server would even persist the triple if it has strong shape validation in place that only persists an ACP graph (which would be perfectly fair and possibly a recommended practice).

I don't want to retrieve all the Access Controls in a graph, then make them into things, then look for one of four made up types and check that it is the one defined within this resource and not an inherited one.

It is orders of magnitude more complex and actually counter-productive for no practical benefit.

When and if we get rid of default access controls, the names of pre-existing access controls will not matter and code that wants to rely on pre-existing defaults will still be able to deal with it at no added complexity.

Reply:
Not even mentioning that there is currently no good support for external vs local access controls and I/we don't have bandwidth to solve that now.